### PR TITLE
Interface for store uploads

### PR DESCRIPTION
**Pull request by pranavrc**
Plus a DELETE flag

### CODE REVIEW

File: storeinterface/storeinterface.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
+package storeinterface
```
Comment (d2lam) [inline storeinterface/storeinterface.go:1]: where is this used?

File: storeinterface/storeinterface.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	token  string
+	client *http.Client
+}
+
+// NewFileUploader returns an StoreInterface for a given url.
+func NewFileUploader(url, token string) StoreInterface {
+	return &sdUploader{
+		url,
+		token,
+		&http.Client{Timeout: 30 * time.Second},
```
Comment (d2lam) [inline storeinterface/storeinterface.go:35]: not sure if we need all this?
I already integrated the httpClient into the `store-cli.go` itself. Don't think we need this whole sdUploader?

Comment (pranavrc) [reply storeinterface/storeinterface.go]: `store-cli.go` doesn't seem to have the logic to upload the contents of a file so far. It's also not using the `val` parameter you pass to the `set` function. Have you pushed all your changes?

File: storeinterface/storeinterface.go (hunk 1 above)
Comment (pranavrc) [reply storeinterface/storeinterface.go]: It's similar to the store uploader in log-service except without the `makeURL` stuff. We can use it in `store-cli.go` to serialize and upload files to the store I guess?

**Review by d2lam — APPROVED** [review]
ok let's merge this and move things out of store-cli.go

**Pull request merged**